### PR TITLE
polish(pu): polish petting_zoo env and config

### PR DESCRIPTION
**Pull request by puyuan1996**
polish petting_zoo env to be compatiable with shared_memory=True case and polish petting_zoo config.

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: dizoo/petting_zoo/config/ptz_simple_spread_atoc_config.py [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 n_landmark = n_agent  # In simple_spread_v2, n_landmark must = n_agent
-collector_env_num = 4
-evaluator_env_num = 5
+collector_env_num = 8
+evaluator_env_num = 8
 communication = True
 ptz_simple_spread_atoc_config = dict(
     exp_name='ptz_simple_spread_atoc_seed0',
     env=dict(
+        manager=dict(shared_memory=False, ),  # NOTE
```
Comment (PaParaZz1) [inline dizoo/petting_zoo/config/ptz_simple_spread_atoc_config.py:11]: why add shm and `NOTE` here

File: ding/envs/env_manager/subprocess_env_manager.py [hunk 2]
```diff
@@ -240,11 +240,15 @@ def _create_state(self) -> None:
         self._reset_param = {i: {} for i in range(self.env_num)}
         if self._shared_memory:
             obs_space = self._observation_space
+            # for env multiagent_mujoco and petting_zoo
```
Comment (PaParaZz1) [inline ding/envs/env_manager/subprocess_env_manager.py:243]: `# for multi_agent case, such as multiagent_mujoco and petting_zoo mpe` can be better

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: dizoo/petting_zoo/config/ptz_simple_spread_atoc_config.py (hunk 1 above)
Comment (puyuan1996) [reply dizoo/petting_zoo/config/ptz_simple_spread_atoc_config.py]: 这个config里面设置了agent_obs_only=True, obs与通常的不一样，现在改好了哈。

File: ding/envs/env_manager/subprocess_env_manager.py (hunk 2 above)
Comment (puyuan1996) [reply ding/envs/env_manager/subprocess_env_manager.py]: 好的

File: dizoo/petting_zoo/config/ptz_simple_spread_atoc_config.py (hunk 1 above)
Comment (Will-Nie) [reply dizoo/petting_zoo/config/ptz_simple_spread_atoc_config.py]: 话说请问这个NOTE对应的解释被放在哪里了呢？

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**